src/repositories/user.repository.ts

```diff
@@ -7,14 +8,25 @@ export const getAdminUser = () => {
 };
 
 export const findUserByUserName = async (username: string) => {
-  return UserModel.findOne({ userName: username });
+  return UserModel.findOne({ username: username });
+};
+
+export const findUserByUserNameAndSelect = async (
```

Comment:
why it should be in a separate method? is it reusing somewhere else except `findUserData`?